EXOSIMS/Observatory/TwoStarShades_mission.py

```diff
@@ -0,0 +1,567 @@
+                ), "At least one slewTime is nan"
+
+            return slewTimes
+
+        if obsTimes == 1:
+            self.ao[:, 1] = self.thrust / self.scMass[:, 1]
+            slewTime_fac = (
+                (
+                    2
+                    * self.occulterSep
+                    / np.abs(self.ao[:, 1])
+                    / (self.defburnPortion / 2.0 - self.defburnPortion**2.0 / 4.0)
+                )
+                .decompose()
+                .to("d2")
+            )
+
+            if old_sInd is None:
+                slewTimes = np.zeros(TL.nStars) * u.d
+            else:
+                # calculate slew time
+                slewTimes = np.sqrt(
+                    slewTime_fac * np.sin(abs(sd) / 2.0)
+                )  # an issue exists if sd is negative
+
+                # The following are debugging
+                assert (
+                    np.where(np.isnan(slewTimes))[0].shape[0] == 0
+                ), "At least one slewTime is nan"
+
+            return slewTimes
+
+    def calculate_dV(self, TL, old_sInd, sInds, sd, slewTimes, tmpCurrentTimeAbs):
+        """Finds the change in velocity needed to transfer to a new star line of sight
+
+        This method sums the total delta-V needed to transfer from one star
+        line of sight to another. It determines the change in velocity to move from
+        one station-keeping orbit to a transfer orbit at the current time, then from
+        the transfer orbit to the next station-keeping orbit at currentTime + dt.
+        Station-keeping orbits are modeled as discrete boundary value problems.
+        This method can handle multiple indeces for the next target stars and calculates
+        the dVs of each trajectory from the same starting star.
+
+        Args:
+            dt (float 1x1 ndarray):
+                Number of days corresponding to starshade slew time
+            TL (float 1x3 ndarray):
+                TargetList class object
+            nA (integer):
+                Integer index of the current star of interest
+            N  (integer):
+                Integer index of the next star(s) of interest
+            tA (astropy Time array):
+                Current absolute mission time in MJD
+
+        Returns:
+            float nx6 ndarray:
+                State vectors in rotating frame in normalized units
+        """
+
+        if old_sInd is None:
+            dV = np.zeros(slewTimes.shape)
+        else:
+            print(slewTimes)
+            dV = np.zeros(slewTimes.shape)
+            slewTimes = slewTimes.reshape(1, 1)
+            badSlews_i, badSlew_j = np.where(slewTimes < self.occ_dtmin.value)
+            for i in range(len(sInds)):
+                for t in range(len(slewTimes.T)):
+                    dV[i, t] = self.dV_interp(slewTimes[i, t], sd[i].to("deg"))
+            dV[badSlews_i, badSlew_j] = np.Inf
+
+        return dV * u.m / u.s
+
+    def minimize_slewTimes(self, TL, nA, nB, tA):
```

Comment:
i thought you weren't using this functionality?  Does it actually work?

Reply:
It doesn't, deleted this method from this module. Fixed